nifi-registry-core/nifi-registry-jetty/src/main/java/org/apache/nifi/registry/jetty/JettyServer.java

```diff
@@ -301,6 +338,15 @@ private WebAppContext loadWar(final File warFile, final String contextPath, fina
         // configure the max form size (3x the default)
         webappContext.setMaxFormContentSize(600000);
 
+        // add HTTP security headers to all responses
+        final String ALL_PATHS = "/*";
+        ArrayList<Class<? extends Filter>> filters = new ArrayList<>(Arrays.asList(XFrameOptionsFilter.class, ContentSecurityPolicyFilter.class, XSSProtectionFilter.class));
+        if(properties.isHTTPSConfigured()) {
+            filters.add(StrictTransportSecurityFilter.class);
+        }
+
+        filters.forEach( (filter) -> addFilters(filter, ALL_PATHS, webappContext));
```

Comment:
Should we be somehow skipping over the webApiContext since we are handling that a different way?

I suppose it doesn't really matter, but I think we end up with the web-api having both sets of filters, but never reaching these since they end up after Jersey.

Reply:
I didn't check what the context looks like after using the spring security settings, but it doesn't appear to affect much using both. I can skip over it if you think that's better.